### PR TITLE
Require Jenkins 2.440.3 or newer

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Require Jenkins 2.440.3 or newer

[Installation statistics](https://stats.jenkins.io/pluginversions/promoted-builds.html) show that 85% of the installations of the most recent release (delivered 3 months ago) are already running Jenkins 2.440.3 or newer.  Users that are upgrading to the most recent release of the plugin are also upgrading to the most recent Jenkins releases.

Removes unnecessary declarations of mockito-core and hamcrest since those are provided by the parent pom.

Removes unnecessary exclusions from other dependencies.

Annotates deprecated methods to silence compiler warnings

Replace rebuild web action in test with freestyle build

Intentionally sacrifices some code coverage of the rebuild plugin so that I don't need to find an alternate way to invoke rebuild.  The previously tested web action is intentionally disallowed by SECURITY-3033 as implemented in:

* https://github.com/jenkinsci/rebuild-plugin/pull/155

### Your checklist for this pull request

- [x] Pull request title represents the changelog entry which will be used in Release Notes. JIRA issue is a part of the title (see existing PRs as examples)
- [x] Please describe what you did. Short summary for reviewers. If the change involves WebUI, add screenshots 
- [x] Link to relevant [Jenkins JIRA issues](https://issues.jenkins-ci.org) or pull requests
- [x] Test automation, if relevant. We expect autotests for all new features or complex bugfixes
- [x] Documentation if relevant (new features). We want to use Documentation-as-Code, just put docs to README or to new Doc files
- [x] Javadoc for new APIs. Any public/protected method is considered as API unless annotated by `Restricted` ([docs](https://kohsuke.org/access-modifier/apidocs/org/kohsuke/accmod/AccessRestriction.html)) 

### CC

@basil

### CODE REVIEW

File: src/test/java/hudson/plugins/promoted_builds/PromotedBuildRebuildParameterProviderTest.java [hunk 1]
```diff
@@ -81,11 +81,7 @@ public void testRebuild() throws Exception {
         Assert.assertEquals(b1.getNumber(), pbpv1.getRun().getNumber());
 
         // rebuild it
-        JenkinsRule.WebClient wc = j.createWebClient();
-        HtmlPage page = wc.getPage(b2, "rebuild");
-        HtmlForm form = page.getFormByName("config");
-        j.submit(form);
-        j.waitUntilNoActivity();
+        FreeStyleBuild b3 = j.assertBuildStatusSuccess(p2.scheduleBuild2(0));
```
Comment (basil) [inline src/test/java/hudson/plugins/promoted_builds/PromotedBuildRebuildParameterProviderTest.java:84]: This is defeating the purpose of the test, which exists solely to exercise rebuilder functionality.

Rather, the test should be adapted to work with POST requests. Try calling `JenkinsRule.WebClient#.withThrowExceptionOnFailingStatusCode(false)`.

**Review by basil — CHANGES_REQUESTED** [review]

File: src/test/java/hudson/plugins/promoted_builds/PromotedBuildRebuildParameterProviderTest.java (hunk 1 above)
Comment (basil) [reply src/test/java/hudson/plugins/promoted_builds/PromotedBuildRebuildParameterProviderTest.java:84]: > Try calling `JenkinsRule.WebClient#withThrowExceptionOnFailingStatusCode(false)`.

Actually the trick is probably to follow https://github.com/jenkinsci/rebuild-plugin/pull/155 in calling `WebClient.createCrumbedUrl` with `HttpMethod.POST`.

Comment (MarkEWaite) [reply src/test/java/hudson/plugins/promoted_builds/PromotedBuildRebuildParameterProviderTest.java]: Thanks for the pointer to https://github.com/jenkinsci/rebuild-plugin/pull/155 .  I read it more carefully and saw that in at least one test, it was clicking the "Rebuild" button and then submitting the resulting form.  That seems to work in this case as well.

43e7046c4aa1bf1d4f36e324115ae8a509a6a081 is the proposed addition to the test that clicks the Rebuild button and then submits the resulting form.

**Review by basil — APPROVED** [review]
Thanks!

**Pull request merged**